packages/next/src/shared/lib/normalized-asset-prefix.ts

```diff
@@ -1,11 +1,11 @@
 export function normalizedAssetPrefix(assetPrefix: string | undefined): string {
-  const escapedAssetPrefix = assetPrefix?.replace(/^\/+/, '') || false
-
-  // assetPrefix as a url
-  if (escapedAssetPrefix && escapedAssetPrefix.startsWith('://')) {
-    return escapedAssetPrefix.split('://', 2)[1]
+  // if assetPrefix is a URL, we return the pathname
+  if (assetPrefix?.startsWith('http')) {
```

Comment:
The original implementation did not discriminate on protocols. Could we just leverage `URL.canParse(assetPrefix)` to enter this branch?

Reply:
TIL `URL.canParse` Thank you!

Reply:
It's pretty new so you should double check if it's available in every supported environment. In the end, it's just a convenience wrapper around `try-catch` of the `URL.parse`